language_bindings/rust/rust_runtime/src/runtime_bindings.rs

```diff
@@ -0,0 +1,300 @@
+
+//! runtime_bindings is a manually generated using bindgen on halide_runtime.h.
+//!
+//! Eventually automatically generating this file may be a better approach, but may introduce stability issues.
```

Comment:
FWIW, pre-generated bindings are generally preferred because `bindgen` is a heavy dependency and its process is somewhat fragile. It might be nice to provide a shell script to update it, however.

Reply:
> pre-generated bindings are generally preferred 

That's quite surprising to me. Not the way other binding tools I've worked with, work. It runs the risk of very rapidly getting the bindings out of sync with the main library. But trusting that this is the case...

> It might be nice to provide a shell script to update it, however.

...I think this is mandatory. At the very least, the command used to create this file needs to be documented _somewhere_.

Reply:
Life sucks for a user who wants to compile something with FFI bindings, only to find that they need to install a suitable version of libclang. Particularly on Windows this can be a high bar, but even on Linux it can be very painful if e.g. a suitable version is not packaged. Given that the output doesn't vary between user machines for most sensible C APIs anyway, it makes more sense for it to be the maintainer's responsibility to keep it up to date.

It can be very helpful to use CI to assert that the generated bindings are up to date by rerunning the script and checking for a diff in the output. Particularly apt in this case, where there may not be a dedicated maintainer for the Rust bindings.

Reply:
> It can be very helpful to use CI to assert that the generated bindings are up to date by rerunning the script and checking for a diff in the output. Particularly apt in this case, where there may not be a dedicated maintainer for the Rust bindings.

Yeah, I think this is sufficient.